Manual/RecursiveDefs.lean

```diff
@@ -180,16 +180,111 @@ After the first step of elaboration, in which definitions are still recursive, a
 tag := "partial-unsafe"
 %%%
 
-:::planned 59
-This section will describe `partial` and `unsafe` definitions:
 
+While most Lean functions can be reasoned about in Lean's type theory as well as compiled and run, definitions marked {keyword}`partial` or {keyword}`unsafe` cannot be meaningfully reasoned about.
+From the perspective of the logic, {keyword}`partial` functions are opaque constants and {keyword}`unsafe` definitions do not exist.
+In exchange for the inability to use these functions for reasoning, there are far fewer requirements placed on them; this can make it possible to write programs that would be impractical or cost-prohibitive to prove anything about, while not giving up formal reasoning for the rest.
+In essence, the {keyword}`partial` subset of Lean is a traditional functional programming language that is nonetheless deeply integrated with the theorem proving features, and the {keyword}`unsafe` subset features the ability to break Lean's runtime invariants if needed, at the cost of less integration with Lean's theorem-proving features.
+Analogously, {keyword}`noncomputable` definitions may use features that don't make sense in programs, but are meaningful in the logic.
 
- * Interaction with the kernel and elaborator
- * What guarantees are there, and what aren't there?
- * How to bridge from unsafe to safe code?
+## Partial Functions
 
+The {keyword}`partial` modifier may only be applied to function definitions.
+Partial functions are not required to demonstrate termination, and Lean does not attempt to do so.
+These functions are “partial” in the sense that they do not necessarily specify mapping from each element of the domain to an element of the codomain, because they might fail to terminate for some or all elements of the domain.
```

Comment:
```suggestion
These functions are “partial” in the sense that they do not necessarily specify a mapping from each element of the domain to an element of the codomain, because they might fail to terminate for some or all elements of the domain.
```

Reply:
We crossed messages on this one! Thanks though!